### PR TITLE
Add support for restore sources and package version props url

### LINKED ISSUE #654: Implement support for controlling the nuget feeds are used programmatically
In addition to #651, a universe build needs to be able to build with additionally specified NuGet feeds

### LINKED ISSUE #656: Implement configurable dependencies versions
Allow configuring the versions of external packages in build/dependencies.props from an external PackageVersionsPropsPath.

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #654
Fixes #656

### CODE REVIEW

File: build/repo.targets [hunk 1]
```diff
@@ -92,20 +92,22 @@
...
+      <_RestoreSources Include="$(BuildDir)" />
+      <_RestoreSources Include="$(IntermediateExternalPackageDir)" />
+      <_RestoreSources Include="$(IntermediateMirrorPackageDir)" />
     </ItemGroup>
 
     <GeneratePackageVersionPropsFile
       Packages="@(_LineupPackages)"
-      OutputPath="$(GeneratedPackageVersionPropsPath)" />
+      OutputPath="$(GeneratedPackageVersionPropsPath)"
+      AdditionalImports="$(DotNetPackageVersionPropsPath)" />
```
Comment (natemcmaster) [inline build/repo.targets]: Not currently a parameter on this task. Does this require an update to buildtools?

File: run.sh [hunk 2]
```diff
@@ -215,9 +233,23 @@ if [ -f "$config_file" ]; then
...
+if [ "$package_version_props_url" ]; then
+    intermediate_dir="$repo_path/obj"
+    props_file_path="$intermediate_dir/external-dependencies.props"
+    mkdir -p "$intermediate_dir"
+    __get_remote_file "$package_version_props_url" "$props_file_path"
+    msbuild_args+="-p:DotNetPackageVersionPropsPath=\"$props_file_path\""
+fi
+
+if [ "$restore_sources" ]; then
+    msbuild_args+="-p:DotNetRestoreSources=\"$restore_sources\""
```
Comment (natemcmaster) [inline run.sh]: Add a space at the end

File: run.sh [hunk 3]
```diff
@@ -215,9 +233,23 @@ if [ -f "$config_file" ]; then
...
+fi
+
 [ -z "$channel" ] && channel='dev'
 [ -z "$tools_source" ] && tools_source='https://aspnetcore.blob.core.windows.net/buildtools'
 
 get_korebuild
 set_korebuildsettings "$tools_source" "$DOTNET_HOME" "$repo_path" "$config_file"
-invoke_korebuild_command "$command" "$@"
+
+echo "$msbuild_args"
```
Comment (natemcmaster) [inline run.sh]: Remove or condition this on "--verbose"

**Comment by pranavkm** [comment]
🆙 📅

File: build/repo.targets (hunk 1 above)
Comment (natemcmaster) [reply build/repo.targets]: Can you commit the update to korebuild-lock.txt required to acquire https://github.com/aspnet/BuildTools/commit/166e381b9436b5c82434af3e53a5ffb63092e525?

**Comment by natemcmaster** [comment]
Once korebuild-lock.txt is updated, :shipit:

**Pull request merged**